doc/AST_FORMAT.md

```diff
@@ -1067,6 +1067,31 @@ However, the following code results in a parse error:
 def f(*a: b); end
 ~~~
 
+## Numbered arguments
+
+Ruby 2.7 introduced a feature called "numbered parameters". This AST node
+doesn't have any sourcemaps and its purpose is to indicate the amount of
```

Comment:
That's troublesome, as all of our exisiting nodes have a source map with at least `expr`...

Reply:
Is there any other way to attach metadata to the node? Can we add a `metadata` attribute to the `AST::Node`? I guess that's a breaking change, libraries that use `parser` may define this field on their own, they'll have to introduce some changes to support it.

Another option could be introducing a new method `parse_with_metadata` that could return `[ast, node_to_its_metadata_map]`. Sounds ugly, that's just an idea.

It's all about attaching arbitrary metadata to a single node (I agree that this metadata can't be represented by an artificial node)

Reply:
I think this should be a different node. Perhaps instead of `block` there should be a `numblock` node that would have the number of parameters instead of a `args` child node.

Reply:
Like this?
```
s(:numblock,
  send_or_lambda_node,
  max_numparam,
  body_node)
```

Reply:
Yes.

Reply:
Done, PTAL